Formula/llvm.rb

```diff
@@ -108,33 +128,54 @@ class Llvm < Formula
   keg_only :provided_by_osx
 
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
-  option "with-clang-extra-tools", "Build extra tools for Clang"
-  option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
-  option "with-lld", "Build LLD linker"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
+  option "without-clang-extra-tools", "Do not build extra tools for Clang"
+  option "without-compiler-rt", "Do not build Clang runtime support libraries for code sanitizers, builtins, and profiling"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
+  option "without-lld", "Do not build LLD linker"
   option "with-lldb", "Build LLDB debugger"
+  option "without-openmp", "Build without the additional OpenMP Runtime Libraries"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
-  option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "without-rtti", "Build without C++ RTTI"
+  option "without-utils", "Do not install utility binaries"
+  option "without-polly", "Build without Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build shared instead of static libraries"
+  option "without-libffi", "Use libffi to call external functions from the interpreter"
+  option "with-all-targets", "Build all targets rather than just AMDGPU, ARM, NVPTX, and X86"
+
+  depends_on "libffi" => :recommended # llvm.org/docs/GettingStarted.grml#requirements
+  depends_on "graphviz" => :optional # for the 'dot' tool (lldb)
+  depends_on "ocaml" => :optional
+
+  if build.with? "python"
+    depends_on "python"
```

Comment:
@xu-cheng your issue should be fixed in the latest commit. However, more changes might be necessary if we decide to change the behavior of the "with-python" option.

What happened in your case was that `lldb` installer found system's python, system's python libraries and system's `Python.h` header. I've changed it so that if lldb is built with python, then libraries and Python.h is determined by `python-config` script. However, I've made it so that if no python is requested then python stuff is not built at all. And this is where we can change things: instead, we can build against system's python/libs/header file. I need feedback/suggestions here.